src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -406,16 +406,29 @@ public bool ContainsFile(string moniker)
             }
         }
 
-        public IVisualStudioHostDocument GetCurrentDocumentFromPath(string filePath)
+        public IVisualStudioHostDocument GetCurrentDocumentFromPath(string filePath, bool additionalFile)
         {
             lock (_gate)
             {
-                return _documentMonikers.TryGetValue(filePath, out var value)
-                    ? value.document
-                    : null;
+                foreach (var document in _documentMonikers[filePath])
+                {
+                    if (document.Key.IsAdditionalFile == additionalFile)
+                    {
+                        return document;
+                    }
+                }
+
+                return null;
             }
         }
 
+        [Obsolete("Call the overload that specifies additionalFile instead.", error: true)]
+        public IVisualStudioHostDocument GetCurrentDocumentFromPath(string filePath)
+        {
+            // Originally this method would return a host document for either, so preserve that until it is removed.
+            return GetCurrentDocumentFromPath(filePath, additionalFile: false) ?? GetCurrentDocumentFromPath(filePath, additionalFile: true);
+        }
```

Comment:
Why are we making a public method on an internal class as `Obsolete` with `error: true`? This is already going to break builds, so what's the harm in deleting it? What's the plan for deleting it at a later date?